jormungandr/src/blockchain/storage.rs

```diff
@@ -37,205 +113,306 @@ struct BlockIterState {
     pending_infos: Vec<BlockInfo<HeaderHash>>,
 }
 
-impl Storage {
+impl Storage03 {
     pub fn new(storage: NodeStorage) -> Self {
-        Storage {
-            read_connection: storage.clone(),
-            write_connection_lock: Lock::new(storage),
-        }
+        let mut rt = runtime::Builder::new()
+            .name_prefix("new-storage-worker-")
+            .core_threads(1)
+            .build()
+            .unwrap();
+
+        rt.block_on_std(async move {
```

Comment:
I guess we only initialize the storage once in the synchronous init code. I vaguely remember there is a one-stop-shop function for this in Tokio, so that you don't have to build a runtime explicitly.

Reply:
I didn't find such function in Tokio. There is `block_on` in `futures`, but it's not applicable here, because bb8 depends on Tokio runtime timer.